### ISSUE TITLE
The external table's LOCATION attribute can't be read from catalog correctly

### ISSUE THREAD

**Issue by water32** [greenplum-db/gpdb#17179]
### Greenplum version or build

Greenplum 7.1

### Expected behavior

Works like previous version instead of error

### Actual behavior

Report an error

### Step to reproduce the behavior
In version 7：
```
$ echo "1" > /tmp/\|.txt
DROP EXTERNAL TABLE if exists public.test_ext;

CREATE READABLE EXTERNAL TABLE public.test_ext
(
  id int
)
LOCATION(
  'file://gp7/tmp/|.txt'
)
FORMAT 'TEXT' ( delimiter 'off')
ENCODING 'UTF8';
select * from public.test_ext;
ERROR:  invalid URI '.txt' : undefined structure
```
In version 6：
```
$ echo "1" > /tmp/\|.txt
DROP EXTERNAL TABLE if exists public.test_ext;

CREATE READABLE EXTERNAL TABLE public.test_ext
(
  id int
)
LOCATION(
  'file://gp7/tmp/|.txt'
)
FORMAT 'TEXT' ( delimiter 'off')
ENCODING 'UTF8';
select * from public.test_ext;
 id
----
  1
(1 row)
```
In the function transformLocationUris The List *locs was connected to long string delemited by pipe(|),  the LOCATION attribute can be stored  multitimes in pg_foreign_table.ftoptions or be escaped.

**Comment by water32** [comment]
After create external table, query the catalog, one location attribute is delimited as 2 records.
```
# SELECT urilocation from pg_exttable WHERE reloid = 'public.test_ext'::regclass;
      urilocation
------------------------
 {file://gp7/tmp/,.txt}
(1 row)
```

**Comment by water32** [comment]
I also submitted a PR to fix this issue：https://github.com/greenplum-db/gpdb/pull/17199

**Comment by jingwen-yang-yjw** [comment]
> I also submitted a PR to fix this issue：#17199

Hi @water32 ,
I noticed that https://github.com/greenplum-db/gpdb/pull/17199 you submitted and https://github.com/greenplum-db/gpdb/pull/17196 I submitted actually have the same solution to this issue.
Using "||" to replace "|" in location uri and recognize it while tokenizing location_uris.

Would you mind if PR https://github.com/greenplum-db/gpdb/pull/17196 is remained and co-authored-by you?

**Comment by water32** [comment]
> > I also submitted a PR to fix this issue：#17199
> 
> Hi @water32 , I noticed that #17199 you submitted and #17196 I submitted actually have the same solution to this issue. Using "||" to replace "|" in location uri and recognize it while tokenizing location_uris.
> 
> Would you mind if PR #17196 is remained and co-authored-by you?

I recommend keeping easy understanded code，and don't care about being a co-author

**Comment by jingwen-yang-yjw** [comment]
> > > I also submitted a PR to fix this issue：#17199
> > 
> > 
> > Hi @water32 , I noticed that #17199 you submitted and #17196 I submitted actually have the same solution to this issue. Using "||" to replace "|" in location uri and recognize it while tokenizing location_uris.
> > Would you mind if PR #17196 is remained and co-authored-by you?
> 
> I recommend keeping easy understanded code，and don't care about being a co-author

OK. Have closed PR https://github.com/greenplum-db/gpdb/pull/17196.
Let's work with PR https://github.com/greenplum-db/gpdb/pull/17199.

**Comment by jingwen-yang-yjw** [comment]
This issue is fixed by https://github.com/greenplum-db/gpdb/pull/17199. So close it here.
